### PR TITLE
v.1.0 of validated PCIe devices with YADRO

### PR DESCRIPTION
**Pull request by staralex86**

### CODE REVIEW

File: pci.ids [hunk 1]
```diff
@@ -1,225 +1,110 @@
...
-#
-
-# Vendors, devices and subsystems. Please keep sorted.
-
-# Syntax:
-# vendor  vendor_name
-#	device  device_name				<-- single tab
-#		subvendor subdevice  subsystem_name	<-- two tabs
-
-ffff  Illegal Vendor ID
```
Comment (alatarum) [inline pci.ids:12]: Please, keep header and footer (class list), insert your data here.

File: pci.ids [hunk 2]
```diff
@@ -1,225 +1,110 @@
...
+
+#ETH Cards
+
+
+1000  Broadcom / LSI
+	16d7  BCM57414 NetXtreme-E 10Gb/25Gb RDMA Ethernet Controller
+		14e4 1402  BCM957414A4142CC 10Gb/25Gb Ethernet PCIe
+
+8086  Intel Corporation
+        1533  I210 Gigabit Network Connection
```
Comment (alatarum) [inline pci.ids]: No spaces before dev id, only tabs allowed here.

File: pci.ids [hunk 3]
```diff
@@ -1,225 +1,110 @@
...
+#SAS/SATA RAID/HBA controllers
+#-----------------------------
+
+1044  Adaptec
+1000  Broadcom / LSI
+	00ce  MegaRAID SAS-3 3316 [Intruder]
+		1000 9380  MegaRAID SAS 9380-8e
+		1000 9390  MegaRAID SAS 9380-8i8e
+	10e2  MegaRAID 12GSAS/PCIe Secure SAS39xx
+                1000 4010  MegaRAID 9560-8i
```
Comment (alatarum) [inline pci.ids]: before subsystem only tabs too

File: pci.ids [hunk 4]
```diff
@@ -1,225 +1,110 @@
...
+		1137 013b  Ethernet Converged NIC X710-DA4
+		1590 0000  Ethernet Controller X710 for 10GbE SFP+
+	1593  Ethernet Controller E810-C for SFP
+		1137 02be  E810XXVDA2 2x25/10 GbE SFP28 PCIe NIC
+
+15b3  Mellanox Technologies
+	1015  MT27710 Family [ConnectX-4 Lx]
+		15b3 0003  Stand-up ConnectX-4 Lx EN, 25GbE dual-port SFP28, PCIe3.0 x8, MCX4121A-ACAT
+	101d  MT2892 Family [ConnectX-6 Dx]
+1374  Silicom Ltd.
```
Comment (alatarum) [inline pci.ids]: Only vendor, without devices? What is this for in the "list of approved devices"?

File: pci.ids [hunk 5]
```diff
@@ -1,225 +1,110 @@
...
+	1521  I350 Gigabit Network Connection
+        37d0  Ethernet Connection X722 for 10GbE SFP+
+	1572  Ethernet Controller X710 for 10GbE SFP+
+		1137 020a  Ethernet Converged NIC X710-DA2
+		1137 013b  Ethernet Converged NIC X710-DA4
+		1590 0000  Ethernet Controller X710 for 10GbE SFP+
+	1593  Ethernet Controller E810-C for SFP
+		1137 02be  E810XXVDA2 2x25/10 GbE SFP28 PCIe NIC
+
+15b3  Mellanox Technologies
```
Comment (alatarum) [inline pci.ids]: Can you at least keep it sorted within sections?

File: pci.ids [hunk 6]
```diff
@@ -1,225 +1,110 @@
...
+		1028 1ff5  Express Flash PM1725b 6.4TB SFF
+		1028 1ff6  Express Flash PM1725b 12.8TB SFF
+1c58  HGST, Inc.
+	0023  Ultrastar SN200 Series NVMe SSD
+		1c58 8823  Ultrastar Memory (ME200)
+1b96  Western Digital
+	2400  Ultrastar DC SN640 NVMe SSD
+	2401  Ultrastar DC SN640 NVMe SSD
+	2402  Ultrastar DC SN640 NVMe SSD
+	2404  Ultrastar DC SN640 NVMe SSD
```
Comment (alatarum) [inline pci.ids]: Are they all same? I guess, should have different size (model).

File: pci.ids [hunk 7]
```diff
@@ -1,225 +1,110 @@
...
+		1000 4000  MegaRAID 9560-16i
+	005d  MegaRAID SAS-3 3108 [Invader]
+		1000 9361  MegaRAID SAS 9361-8i
+		1000 9363  MegaRAID SAS 9361-4i
+                1000 9371  MegaRAID SAS 9361-16i
+                
+	0090  SAS3108 PCI-Express Fusion-MPT SAS-3
+	0091  SAS3108 PCI-Express Fusion-MPT SAS-3
+	0094  SAS3108 PCI-Express Fusion-MPT SAS-3
+	0095  SAS3108 PCI-Express Fusion-MPT SAS-3
```
Comment (alatarum) [inline pci.ids]: Are they same "SAS3108"?

File: pci.ids [hunk 8]
```diff
@@ -1,225 +1,110 @@
...
+		10df e300  LPe32002-M2 2-Port 32Gb Fibre Channel Adapter
+#Accelerators
+#------------
+
+10de  NVIDIA Corporation
+      1db6  GV100GL [Tesla V100 PCIe 32GB]
+      1df6  GV100GL [Tesla V100S PCIe 32GB]
+      1eb8  TU104GL [Tesla T4]
+
+1002  Advanced Micro Devices, Inc. [AMD/ATI]
```
Comment (alatarum) [inline pci.ids]: Empty?

**Review by alatarum — CHANGES_REQUESTED** [review]

**Review by AlexanderAmelkin — CHANGES_REQUESTED** [review]
Please address Andrey's comments, I totally second them.

**Comment by staralex86** [comment]
Made changes, sorted devices inside its classes

**Comment by staralex86** [comment]
Made needed changes.

File: pci.ids [hunk 9]
```diff
@@ -9,6 +9,124 @@
...
+
+#SAS/SATA RAID/HBA controllers
+#-----------------------------
+
+1000  Broadcom / LSI
+	00ce  MegaRAID SAS-3 3316 [Intruder]
+		1000 9380  MegaRAID SAS 9380-8e
+		1000 9390  MegaRAID SAS 9380-8i8e
+	10e2  MegaRAID 12GSAS/PCIe Secure SAS39xx
+				1000 4010  MegaRAID 9560-8i
```
Comment (AlexanderAmelkin) [inline pci.ids]: What are these four tabs? There are just three possible levels of indentation as described in the header:

```
# Syntax:
# vendor  vendor_name
#	device  device_name				<-- single tab
#		subvendor subdevice  subsystem_name	<-- two tabs
```

There can be no lines indented by 3 or 4 tabs. Please fix. Also when fixing, please mind the line endings! They MUST be unix-style (LF only), not windows-style (CR-LF).

**Review by AlexanderAmelkin — CHANGES_REQUESTED** [review]

File: pci.ids [hunk 10]
```diff
@@ -9,6 +9,124 @@
...
+		1000 30a0  SAS9300-8e
+	00c9  SAS3216 PCI-Express Fusion-MPT SAS-3
+
+1044  Adaptec
+
+#NVMe SSD
+#--------
+
+1344  Micron Technology Inc
+		51a2  7300 Dual Port U.2 PRO
```
Comment (alatarum) [inline pci.ids]: 1 unwanted tab

File: pci.ids [hunk 11]
```diff
@@ -9,6 +9,124 @@
...
+		1000 9380  MegaRAID SAS 9380-8e
+		1000 9390  MegaRAID SAS 9380-8i8e
+	10e2  MegaRAID 12GSAS/PCIe Secure SAS39xx
+				1000 4010  MegaRAID 9560-8i
+# 9560 16 internal port RAID controller
+		1000 4000  MegaRAID 9560-16i
+	005d  MegaRAID SAS-3 3108 [Invader]
+		1000 9361  MegaRAID SAS 9361-8i
+		1000 9363  MegaRAID SAS 9361-4i
+				1000 9371  MegaRAID SAS 9361-16i
```
Comment (alatarum) [inline pci.ids]: 2 unwanted tabs

File: pci.ids (hunk 9 above)
Comment (alatarum) [inline pci.ids]: 2 unwanted tabs

File: pci.ids [hunk 12]
```diff
@@ -9,6 +9,124 @@
...
+
+1374  Silicom Ltd.
+
+15b3  Mellanox Technologies
+	1015  MT27710 Family [ConnectX-4 Lx]
+		15b3 0003  Stand-up ConnectX-4 Lx EN, 25GbE dual-port SFP28, PCIe3.0 x8, MCX4121A-ACAT
+	101d  MT2892 Family [ConnectX-6 Dx]
+
+8086  Intel Corporation
+		1533  I210 Gigabit Network Connection
```
Comment (alatarum) [inline pci.ids]: 1 unwanted tab

File: pci.ids [hunk 13]
```diff
@@ -9,6 +9,124 @@
...
+1b96  Western Digital
+	2400  Ultrastar DC SN640 NVMe SSD
+	2401  Ultrastar DC SN640 NVMe SSD
+	2402  Ultrastar DC SN640 NVMe SSD
+	2404  Ultrastar DC SN640 NVMe SSD
+
+1c58  HGST, Inc.
+	0023  Ultrastar SN200 Series NVMe SSD
+		1c58 8823  Ultrastar Memory (ME200)
+
```
Comment (alatarum) [inline pci.ids:113]: I prefer to remove most empty lines in the file

File: pci.ids [hunk 14]
```diff
@@ -9,6 +9,124 @@
...
 
+
+#ETH Cards
+#---------
+
+1000  Broadcom / LSI
+	16d7  BCM57414 NetXtreme-E 10Gb/25Gb RDMA Ethernet Controller
+		14e4 1402  BCM957414A4142CC 10Gb/25Gb Ethernet PCIe
+
+1374  Silicom Ltd.
```
Comment (AlexanderAmelkin) [inline pci.ids]: Do we actually need in this list vendors w/o any approved devices?

File: pci.ids [hunk 15]
```diff
@@ -9,6 +9,146 @@
...
+
+#ETH Cards
+#---------
+
+1000  Broadcom / LSI
+	16d7  BCM57414 NetXtreme-E 10Gb/25Gb RDMA Ethernet Controller
+		14e4 1402  BCM957414A4142CC 10Gb/25Gb Ethernet PCIe
+
+<<<<<<< HEAD
+=======
```
Comment (AlexanderAmelkin) [inline pci.ids]: This is a git conflict marker. You should have resolved and removed that before doing `git add`, committing and pushing.

File: pci.ids [hunk 16]
```diff
@@ -9,6 +9,146 @@
...
+		1077 0299  QLE2740 Single Port 32Gb Fibre Channel to PCIe Adapter
+		1077 029a  QLE2742 Dual Port 32Gb Fibre Channel to PCIe Adapter
+	2532  ISP2532-based 8Gb Fibre Channel to PCI Express HBA
+		1077 015d  QLE2562 PCI Express to 8Gb FC Dual Channel
+		1077 015e  QLE2564 PCI Express to 8Gb FC Quad Channel
+
+#Accelerators
+#------------
+
+<<<<<<< HEAD
```
Comment (AlexanderAmelkin) [inline pci.ids]: Looks like there is quite a number of these...

**Review by AlexanderAmelkin — CHANGES_REQUESTED** [review]

File: pci.ids (hunk 16 above)
Comment (AlexanderAmelkin) [reply pci.ids]: I've fixed that for you.

File: pci.ids [hunk 17]
```diff
@@ -9,6 +9,117 @@
...
 #		subvendor subdevice  subsystem_name	<-- two tabs
 
+# ETH Cards
+#---------
+
+1000  Broadcom / LSI
+	16d7  BCM57414 NetXtreme-E 10Gb/25Gb RDMA Ethernet Controller
+		14e4 1402  BCM957414A4142CC 10Gb/25Gb Ethernet PCIe
+14e4  Broadcom Inc. and subsidiaries
+	1751  BCM57504 NetXtreme-E 10Gb/25Gb/40Gb/50Gb/100Gb/200Gb Ethernet
```
Comment (alatarum) [inline pci.ids:19]: This cards are found on Aramis

**Review by alatarum — APPROVED** [review]

File: pci.ids [hunk 18]
```diff
@@ -9,6 +9,117 @@
...
+	0096  SAS3004 PCI-Express Fusion-MPT SAS-3
+	0097  SAS3008 PCI-Express Fusion-MPT SAS-3
+		1000 30a0  SAS9300-8e
+		1000 30e0  SAS9300-8i
+	00c9  SAS3216 PCI-Express Fusion-MPT SAS-3
+
+# NVMe SSD
+#--------
+
+1344  Micron Technology Inc
```
Comment (alatarum) [inline pci.ids:94]: There is Micron device found on Porthos: 0x1344:0x5180 (9100 PRO NVMe SSD)

File: pci.ids [hunk 19]
```diff
@@ -9,6 +9,117 @@
...
+		1077 015e  QLE2564 PCI Express to 8Gb FC Quad Channel
+
+# Accelerators
+#------------
+
+10de  NVIDIA Corporation
+	1db6  GV100GL [Tesla V100 PCIe 32GB]
+	1df6  GV100GL [Tesla V100S PCIe 32GB]
+	1eb8  TU104GL [Tesla T4]
+
```
Comment (alatarum) [inline pci.ids:65]: There is ATI device found on Richelieu 0x1002:0x683f (Radeon HD 7750/8740 / R7 250E)

Comment (AlexanderAmelkin) [reply pci.ids:65]: The fact that it is installed in a development rig doesn't automatically mean that it's validated by QA.

**Review by AlexanderAmelkin — APPROVED** [review]

**Pull request merged**